Separate SecureSourceManager repository fields into input and observed structs

ericpang777 opened the pull request:
### Change description

<!--
Describe what this pull request does.

* If your pull request is to address an open issue, indicate it by specifying the
issue number: 

For example: "Fixes #858"
-->
Fixes #

### Tests you have done

<!--

Make sure you have run "make ready-pr" to run required tests and ensure this PR is ready to review. 

Also if possible, share a bit more on the tests you have done. 

For example if you have updated the pubsubtopic sample, you can share the test logs from running the test case locally.

go test -v -tags=integration ./config/tests/samples/create -test.run TestAll -run-tests pubsubtopic

-->

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

ericpang777 commented:
Branched off #3746

gemmahou commented:
Looks good. Pending updates on roundtrip_test.go and _http.log/

ericpang777 commented:
> Looks good. Pending updates on roundtrip_test.go and _http.log

Done

gemmahou commented:
> > Looks good. Pending updates on roundtrip_test.go and _http.log
> 
> Done, I also removed the empty URI struct, and added CreateTime. UpdateTime remains semi-commented out because I don't think we even output it. But I guess I can add it to the schema?

No harm to add it now if we plan to output the value in the future.

ericpang777 commented:
> > > Looks good. Pending updates on roundtrip_test.go and _http.log
> > 
> > 
> > Done, I also removed the empty URI struct, and added CreateTime. UpdateTime remains semi-commented out because I don't think we even output it. But I guess I can add it to the schema?
> 
> No harm to add it now if we plan to output the value in the future.

Added back. I'll leave it out of the mock still to reflect real behaviour.

gemmahou commented:
/lgtm

yuwenma reviewed (commented):
/approve
